kore/src/Kore/Simplify/Ceil.hs

```diff
@@ -261,13 +245,16 @@ newAxiomCeilSimplifier = CeilSimplifier $ \input -> do
             Condition.top
             (synthesize $ CeilF input)
             (const empty)
-    return (OrPattern.map toCondition evaluation)
+    return (OrPattern.map (Predicate.toMultiAnd . toPredicate) evaluation)
   where
-    toCondition Conditional{term = Top_ _, predicate, substitution} =
-        Conditional{term = (), predicate, substitution}
-    toCondition patt =
+    -- TODO(Ana): probably should parse MultiAnd here
+    toPredicate Conditional{term = Top_ _, predicate, substitution} =
+        Predicate.makeAndPredicate
+            predicate
+            (from @_ @(Predicate _) substitution)
+    toPredicate patt =
```

Comment:
(nit) could use `Condition.isPredicate` and `from @(Condition v) @(Predicate v)` (but fine to keep)

Reply:
We already have to match on the fields of the `Conditional` here, so I think using `Conditional.isPredicate` would be more convoluted (and less clear).